qt/python/mantidqtinterfaces/mantidqtinterfaces/simplescanviewer/rectangle_selection.py

```diff
@@ -25,55 +23,46 @@ def _release(self, event):
         So for now, we use this version even if there is no rational reason I can see for it to be needed.
         """
 
-        # matplotlib is bound to deprecate the eventpress attribute with the release of version 3.7, making it private
-        # In the meantime, it gives off warnings every time it is accessed.
-        # But since this feature does not work correctly at the moment, and we need this fix as a workaround, we just
-        # silence the warnings and hope that this bug will be understood and solved by the time this deprecation
-        # actually comes into effect.
-        # This is not nearly a perennial fix, but it should hold for a year or so.
-
-        with warnings.catch_warnings():
-            warnings.filterwarnings("ignore", category=DeprecationWarning)
-            if not self.interactive:
-                self.to_draw.set_visible(False)
+        if not self._interactive:
+            self.to_draw.set_visible(False)
 
-            # update the eventpress and eventrelease with the resulting extents
-            x1, x2, y1, y2 = self.extents
-            self.eventpress.xdata = x1
-            self.eventpress.ydata = y1
-            xy1 = self.ax.transData.transform([x1, y1])
-            self.eventpress.x, self.eventpress.y = xy1
+        # update the eventpress and eventrelease with the resulting extents
+        x1, x2, y1, y2 = self.extents
+        self._eventpress.xdata = x1
```

Comment:
We'll need to be careful with this moving forward as `_` means they consider it private so we might have to look for a different way of achieving this in the future.